client/components/horizontal-list/index.tsx

```diff
@@ -3,14 +3,24 @@
 /**
  * External dependencies
  */
+import React from 'react';
 import { List } from '@woocommerce/components';
 
 /**
  * Internal dependencies.
  */
 import './style.scss';
 
-const HorizontalList = ( props ) => {
+interface HorizontalListItem {
+	title: string;
+	content: string | React.ReactNode;
+}
+
+interface Props {
+	items: ( false | HorizontalListItem )[];
```

Comment:
Is there a reason we allow the `item` elements to be either `false` or `HorizontalListItem`? If possible I'd prefer to enforce that the items must contain only `HorizontalListItems` instead of allowing any `false` values.

It's easy to fix any errors that may occur from that where this component is used with a `.filter( elem => false !=== elem )`, e.g. `<HorizontalList items={ items.filter( elem => false !== elem ) } />`.

There might be a legitimate reason for making the type like this, but I don't see it by just looking at the PR 🤔

Reply:
Agreed. Although there was already a filter for that – `filter( Boolean )`, TS wouldn't infer it unless I explicitly add `as HorizontalListItem[]` to the filtered array.

Fixed in [5809133](https://github.com/Automattic/woocommerce-payments/pull/6249/commits/580913319dbd7a1a1f5f7157a280a218cee779cd).